scripts/pre_commit_linter.py

```diff
@@ -896,8 +898,129 @@ def _check_html_directive_name(all_files):
+    parser = pyjsparser.PyJsParser()
+    for filename in files_to_check:
+        with open(filename) as f:
+            content = f.read()
+        # Parse the body of the content as nodes.
+        parsed_nodes = parser.parse(content)['body']
+        for parsed_node in parsed_nodes:
+            # Check the type of the node.
+            if parsed_node['type'] != 'ExpressionStatement':
+                continue
+            # Separate the expression part of the node.
+            expression = parsed_node['expression']
+            # Check whether the expression belongs to a directive.
+            if (expression['type'] != 'CallExpression' or (
+                    expression['callee']['type'] != (
+                        'MemberExpression')) or (
+                            expression['callee']['property']['name'] != (
+                                'directive'))):
+                continue
+            # Separate the arguments of the expression.
+            arguments = expression['arguments']
+            # The first argument of the expression is the
+            # name of the directive.
+            if arguments[0]['type'] == 'Literal':
+                directive_name = str(arguments[0]['value'])
+            arguments = arguments[1:]
+            for argument in arguments:
+                # Check the type of an argument.
+                if argument['type'] != 'ArrayExpression':
+                    continue
+                # Separate out the elements for the argument.
+                elements = argument['elements']
+                for element in elements:
+                    # Check the type of an element.
+                    if element['type'] != 'FunctionExpression':
+                        continue
+                    # Separate out the body of the element.
+                    body = element['body']
+                    if body['type'] != 'BlockStatement':
+                        continue
+                    # Further separate the body elements from the body.
+                    body_elements = body['body']
+                    for body_element in body_elements:
+                        # Check if the body element is a return statement.
+                        if (body_element['type'] != 'ReturnStatement' or (
+                                body_element['argument']['type'] != (
+                                    'ObjectExpression'))):
+                            continue
+                        # Separate the properties of the return node.
+                        return_node_properties = (
+                            body_element['argument']['properties'])
+                        # Loop over all the properties of the return node
+                        # to find out the scope key.
+                        for return_node_property in return_node_properties:
+                            # Check whether the property is scope.
+                            is_property_key_an_identifier = (
```

Comment:
actually, no need to have if-statements like this, here's what it should look like:
```python
is_property_key_an_identifier = (return_node_property['key']['type'] == 'Identifier')
```
This is because the `==` operator only returns boolean.

Reply:
Also just a note, we generally name booleans like this: property_key_is_an_identifier. That's because "is" is a verb and starting names with verbs suggests that the thing being named is a function.

Reply:
Okay, understood! Done, thanks!